draft-ietf-quic-http.md

```diff
@@ -201,6 +201,21 @@ additional framing. Note that a request or response without a body will cause
 this stream to be half-closed in the corresponding direction without
 transferring data.
 
+Because the message control stream contains HPACK data which manipulates
+connection-level state, the message control stream MUST NOT be closed with a
+stream-level error.  If an implementation chooses to reject a request with a
+QUIC error code, it MUST trigger a RST_STREAM on the data stream only.  An
+implementation MAY close (FIN) a message control stream without completing a
+full HTTP message if the data stream has been abruptly closed.  Data on message
+control streams MUST be fully consumed, or the connection terminated.
+
+If a message control stream is terminated abruptly by the transport layer (e.g.
+with QUIC_TOO_MANY_OPEN_STREAMS), any frames already sent on that stream MUST
+be re-played on another message control stream as soon as the maximum number of
+streams permits.  If the implementation no longer requires the response to this
+request, it MAY abruptly terminate the corresponding data stream at the same
+time.
```

Comment:
I think that this is easier to handle with: "If a message control stream is terminated by a RST_STREAM *for any reason*, an endpoint MUST terminate the connection with a YOU_SCREWED_UP error code."  Attempting to rescue the situation isn't worth the effort and aside from exceeding concurrent stream limits during 0-RTT.

I don't think that there is any reason that the transport would legitimately need to close a stream on its own.  Even in that case, that should really be done by the application.